.github/workflows/go-test.yml

```diff
@@ -14,9 +14,29 @@ concurrency:
   cancel-in-progress: true
 
 jobs:
+  localstack-s3-setup:
+    services:
+      s3:
+        image: localstack/localstack:s3-latest
+        ports:
+          - "127.0.0.1:4566:4566"
+        environment:
+          - DEBUG=${DEBUG:-0}
+        volumes:
+          - "./s3_local/init_s3.py:/etc/localstack/init/ready.d/init-s3.py"
+
+    steps:
+    - name: Wait for LocalStack to be ready
+      run: |
+        until curl -s http://127.0.0.1:4566/health | grep "\"s3\": \"running\""; do
+          echo "Waiting for LocalStack to start..."
+          sleep 5
+        done
```

Comment:
not sure if this is possible but if `curl` is installed in the localstack container you could do something like this: 

https://github.com/dennis-tra/nebula/blob/13c80a32bc616e07b78601f91258464fbdafba5a/.github/workflows/push_main.yml#L21

If that doesn't work define a timeout here: https://docs.github.com/en/actions/writing-workflows/workflow-syntax-for-github-actions#jobsjob_idstepstimeout-minutes

so that we don't accidentally burn through our GitHub Actions runner minutes